### PR TITLE
feat: move case to offline state (strike-out) (CMC-991)

### PR DESCRIPTION
**Pull request by swalker125**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CMC-991

### Change description ###

New case strikeout scheduler bpmn to find cases where strike out needs to happen (currently this is just move to offline)
New start business process common bpmn (this can be implemented elsewhere in a future ticket)
New case strikeout bpmn for notifications (sends a notification to applicant solicitor 1 and respondent solicitor 1)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by swalker125** [comment]
Sending emails locally (use pretends key so emails not actually sent)

```
2020-11-20T10:29:19.929 INFO  [TopicSubscriptionManager] u.g.h.r.u.s.tasks.handler.BaseExternalTaskHandler External task 'processCaseEvent' started
2020-11-20T10:29:23.519 INFO  [http-nio-4000-exec-1] u.g.h.reform.unspec.controllers.CallbackController Received callback from CCD, eventId: NOTIFY_RESPONDENT_SOLICITOR1_CASE_STRIKE_OUT, callback type: about-to-submit, page id: Optional.empty, version: Optional.empty
2020-11-20T10:29:24.330 INFO  [TopicSubscriptionManager] u.g.h.r.u.s.tasks.handler.BaseExternalTaskHandler External task 'processCaseEvent' finished
2020-11-20T10:29:24.396 INFO  [TopicSubscriptionManager] u.g.h.r.u.s.tasks.handler.BaseExternalTaskHandler External task 'processCaseEvent' started
2020-11-20T10:29:26.524 INFO  [http-nio-4000-exec-2] u.g.h.reform.unspec.controllers.CallbackController Received callback from CCD, eventId: NOTIFY_APPLICANT_SOLICITOR1_CASE_STRIKE_OUT, callback type: about-to-submit, page id: Optional.empty, version: Optional.empty
2020-11-20T10:29:26.756 INFO  [TopicSubscriptionManager] u.g.h.r.u.s.tasks.handler.BaseExternalTaskHandler External task 'processCaseEvent' finished
```

**Comment by swalker125** [comment]
Screenshot 2020-11-20 at 11 22 47
Screenshot 2020-11-20 at 11 22 43
Screenshot 2020-11-20 at 11 22 40

**Review by dharmendrak — DISMISSED** [review]

File: ccd-definition/CaseEvent.json [hunk 1]
```diff
@@ -451,5 +451,50 @@
...
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "UNSPECIFIED_CLAIMS",
+    "ID": "MOVE_CLAIM_TO_STRUCK_OUT",
+    "Name": "Strike out claim",
+    "Description": "Strikes out a claim",
+    "DisplayOrder": 29,
+    "PreConditionState(s)": "AWAITING_CLAIMANT_INTENTION",
+    "PostConditionState": "PROCEEDS_WITH_OFFLINE_JOURNEY",
+    "SecurityClassification": "Public",
+    "CallBackURLAboutToSubmitEvent": "${CCD_DEF_CASE_SERVICE_BASE_URL}/cases/callbacks/about-to-submit",
```
Comment (pd-dudkiewicz) [inline ccd-definition/CaseEvent.json:466]: We need about to start callback here in order to trigger state flow check.

File: src/test/java/uk/gov/hmcts/reform/unspec/handler/callback/notification/ApplicantClaimStruckOutNotificationHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+        @Test
+        void shouldNotifyApplicantSolicitor_whenInvoked() {
+            CaseData caseData = CaseDataBuilder.builder().atStateFullDefence().build();
+            CallbackParams params = CallbackParamsBuilder.builder().of(ABOUT_TO_SUBMIT, caseData).build();
+
+            handler.handle(params);
+
+            verify(notificationService).sendMail(
+                eq(EMAIL),
```
Comment (pd-dudkiewicz) [inline src/test/java/uk/gov/hmcts/reform/unspec/handler/callback/notification/ApplicantClaimStruckOutNotificationHandlerTest.java]: `eq()` wrapping is not required when we verify with specific values

**Review by pd-dudkiewicz — COMMENTED** [review]
Just a reminder that this PR needs to be separated with service code merged before Camunda.

**Review by douglasrice — DISMISSED** [review]

**Review by douglasrice — APPROVED** [review]

**Review by pd-dudkiewicz — APPROVED** [review]
ignore my previous comment, camunda doesn't need to be merged separately for this one

**Pull request merged**